### PR TITLE
Some more improvements for posts with shares

### PR DESCRIPTION
**Pull request by annando**
There is now a new functionality to fetch shared posts. Also the usage of the function `fetchShareAttributes` is reduced.

### CODE REVIEW

File: src/Content/Item.php [hunk 1]
```diff
@@ -680,11 +680,11 @@ public function createSharedBlockByArray(array $item, bool $add_media = false):
 			$shared_content .= '[h3]' . $item['title'] . "[/h3]\n";
 		}
 
-		$shared = ItemModel::getShareArray($item);
+		$shared = $this->getSharedPost($item, ['uri-id', 'uri', 'body', 'title', 'author-name', 'author-link', 'author-avatar', 'guid', 'created', 'plink', 'network']);
 
 		// If it is a reshared post then reformat it to avoid display problems with two share elements
 		if (!empty($shared)) {
-			if (!empty($shared['guid']) && ($encaspulated_share = self::createSharedPostByGuid($shared['guid'], 0, '', $add_media))) {
+			if (($encaspulated_share = $this->createSharedBlockByArray($shared['post'], $add_media))) {
```
Comment (MrPetovan) [inline src/Content/Item.php]: Extra parentheses.
```suggestion
			if ($encaspulated_share = $this->createSharedBlockByArray($shared['post'], $add_media)) {
```

File: src/Factory/Api/Mastodon/Status.php [hunk 2]
```diff
@@ -155,11 +155,9 @@ public function createFromUriId(int $uriId, int $uid = 0): \Friendica\Object\Api
 			$poll = null;
 		}
 
-		$shared = Item::getShareArray($item);
-		if (!empty($shared['guid'])) {
-			$shared_item = Post::selectFirst(['uri-id', 'plink'], ['guid' => $shared['guid']]);
-
-			$shared_uri_id = $shared_item['uri-id'] ?? 0;
+		$shared = DI::contentItem()->getSharedPost($item, ['uri-id']);
```
Comment (nupplaphil) [inline src/Factory/Api/Mastodon/Status.php]: Please replace `DI::contentItem()` call with including the class directly at `__construct(...)` and attaching it to a new `protected $contentItem;` property.

So you can access it like
```suggestion
		$shared = $this->contentItem->getSharedPost($item, ['uri-id']);
```

File: src/Factory/Api/Twitter/Status.php [hunk 3]
```diff
@@ -178,11 +179,9 @@ private function createFromArray(array $item, int $uid, bool $include_entities):
 
 		$friendica_activities = $this->activities->createFromUriId($item['uri-id'], $uid);
 
-		$shared = Item::getShareArray($item);
-		if (!empty($shared['guid'])) {
-			$shared_item = Post::selectFirst(['uri-id', 'plink'], ['guid' => $shared['guid']]);
-
-			$shared_uri_id = $shared_item['uri-id'] ?? 0;
+		$shared = DI::contentItem()->getSharedPost($item, ['uri-id']);
```
Comment (nupplaphil) [inline src/Factory/Api/Twitter/Status.php]: same as above
```suggestion
		$shared = $this->contentItem->getSharedPost($item, ['uri-id']);
```

File: src/Content/Item.php [hunk 4]
```diff
@@ -695,4 +695,38 @@ public function createSharedBlockByArray(array $item, bool $add_media = false):
...
+
+	/**
+	 * Return the shared post from an item array (if the item is shared item)
+	 *
+	 * @param array $item
+	 * @param array $fields
+	 *
+	 * @return array with the shared post
+	 */
+	public function getSharedPost(array $item, array $fields = []): array
```
Comment (nupplaphil) [inline src/Content/Item.php:707]: This sounds like that there are tests possible?

File: src/Content/Text/BBCode.php [hunk 5]
```diff
@@ -1094,7 +1118,7 @@ function (array $attributes) {
 	 */
 	public static function removeSharedData(string $body): string
```
Comment (nupplaphil) [inline src/Content/Text/BBCode.php:1119]: At least this seems like a good place for unit tests
input = `$body`
expected = return value

**Review by nupplaphil — CHANGES_REQUESTED** [review]

**Review by MrPetovan — CHANGES_REQUESTED** [review]

File: src/Factory/Api/Mastodon/Status.php (hunk 2 above)
Comment (annando) [reply src/Factory/Api/Mastodon/Status.php]: Done

File: src/Factory/Api/Twitter/Status.php (hunk 3 above)
Comment (annando) [reply src/Factory/Api/Twitter/Status.php]: Done

File: src/Content/Item.php (hunk 1 above)
Comment (annando) [reply src/Content/Item.php]: Resolved by mostly restoring the old functionality

**Review by MrPetovan — APPROVED** [review]

**Pull request merged**